table/block_based/block_based_table_builder.cc

```diff
@@ -1080,6 +1080,10 @@ void BlockBasedTableBuilder::Add(const Slice& key, const Slice& value) {
                                       r->ioptions.logger);
   } else {
     assert(false);
+    r->SetStatus(Status::InvalidArgument(
```

Comment:
nit: compared to current behavior, this will mark this builder in a non-usable state and all the data entries that have previously been successfully added to the builder are forfeited. While strictly speaking, this invalid entry doesn't really affect those previous data, and the builder could still be used to add and build more entries. Should we just return here, or maybe something like return `TryAgain`, gives the user a chance to handle recoverable error differently?

Reply:
Thanks for the detailed context. That makes sense!